cpp/Platform.Equality/Platform.Equality.Any.h

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#ifndef PLATFORM_EQUALITY_ANY
+#define PLATFORM_EQUALITY_ANY
+
+#include <any>
+#include <functional>
+#include <typeindex>
+#include <concepts>
+#include <stdexcept>
+
+namespace Platform::Equality
+{
+    namespace Internal
+    {
+        template<class TValue, class TEqualityComparer>
+        inline auto ToAnyEqualVisitor(TEqualityComparer&& func)
+        {
+            return std::pair<std::type_index, std::function<bool(const std::any&, const std::any&)>>
```

Comment:
`std::function<bool(const std::any&, const std::any&)>` is used twice in this code file, you can define using for it. I suggest `EqualityComparerFunctionType` as a name.